### ISSUE TITLE
Copy-paste issue in CredentialPasswordPrivate::checkPassword

### ISSUE THREAD

**Issue by tsilia** [cutelyst/cutelyst#386]
Hello.

This looks like a copy-paste issue:
https://github.com/cutelyst/cutelyst/blob/a55c65681f59c1d338155794e7e2123c2f682ba3/Cutelyst/Plugins/Authentication/credentialpassword.cpp#L230-L237
Please note that it's the password itself that is prepended or appended to the password, rather than the pre- or postsalt.

**Comment by tsilia** [comment]
I can now confirm that it's definitely a copy-paste issue.
You can verify this by changing `testauthentication` to use a postsalt, for example.
Change:
https://github.com/cutelyst/cutelyst/blob/a55c65681f59c1d338155794e7e2123c2f682ba3/tests/testauthentication.cpp#L181
to:
```cpp
fooUser.insert(QStringLiteral("password"), CredentialPassword::createPassword(QByteArrayLiteral("123") + QByteArrayLiteral("postsalt"), QCryptographicHash::Sha256, 10, 10, 10));
```
(append a postsalt to the password)
and this:
https://github.com/cutelyst/cutelyst/blob/a55c65681f59c1d338155794e7e2123c2f682ba3/tests/testauthentication.cpp#L191-L192
to:
```cpp
hashedPassword->setPasswordType(CredentialPassword::Hashed);
hashedPassword->setPasswordPostSalt(QStringLiteral("postsalt"));   // <---------------------- added this line
auth->addRealm(new AuthenticationRealm(hashedStore, hashedPassword, QStringLiteral("hashed")));
```
(set a postsalt to use in `hashedPassword`).
This will break the `auth-user-realm-test00` (among few others), but if you fix the aforementioned copy-paste issue, the test will pass OK.

**Comment by dantti** [comment]
fixed with f7f671479e90f71d8f58641b49c3c5c3eeb2371a
